### PR TITLE
Remove const from hostreadwrite [artv3/hostreadwrite-fix]

### PR DESCRIPTION
**Pull request by artv3**
I'm not sure we want the const since we would like to modify contents. 
```
double *  data = mfem::HostReadWrite(....) 
data[0] += 1.0;
 ```

### CODE REVIEW

**Comment by artv3** [comment]
I think the HostWrite method should also not have const, but I would like to double check with the MFEM team.

**Comment by artv3** [comment]
This PR is ready to go. The gpu/appfixes has other fixes.

File: general/device.hpp [hunk 1]
```diff
@@ -283,9 +283,9 @@ inline T *ReadWrite(Memory<T> &mem, int size, bool on_dev = true)
    }
 }
 
-/** @brief Shortcut to ReadWrite(const Memory<T> &mem, int size, false) */
+/** @brief Shortcut to ReadWrite(Memory<T> &mem, int size, false) */
 template <typename T>
-inline const T *HostReadWrite(const Memory<T> &mem, int size)
+inline T *HostReadWrite(Memory<T> &mem, int size)
```
Comment (camierjs) [inline general/device.hpp]: Shouldn't we do this too line 264 for the HostWrite?

Comment (v-dobrev) [reply general/device.hpp]: `HostWrite` is fixed in #937.

The fix in this PR could have been made there as well.

**Comment by v-dobrev** [comment]
@tzanio, I'll merge this in `next` for testing -- you can take a look in the meantime -- it's a small change.

**Pull request merged**